[14.0][MIG] l10n_fr_fec_oca

alexis-via opened the pull request.

alexis-via committed:
Check-in of the l10n_fr_fec module from github.com/odoo/odoo 12.0 branch under the name l10n_fr_fec_oca

This module was initially developped by Alexis de Lattre in OCA/l10n-france for v8, then moved to odoo core and received several contribution/improvements by Odoo SA employees and it also received fixes and improvements by community members.

alexis-via committed:
Improve FEC generation: encoding, field delimiter, partner export are now configurable

Fix filename when VAT number contrains spaces
PEP8

oca-travis committed:
[UPD] Update l10n_fr_fec_oca.pot

bealdav committed:
FIX non compliant code with oca rules

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

alexis-via committed:
l10n_fr_fec_oca: restore working code

alexis-via committed:
[MIG] l10n_fr_fec_oca to v14

alexis-via committed:
l10n_fr_fec_oca: make my hack for 120000/129000 work on a well-configured DB too

fmdl commented on l10n_fr_fec_oca/wizard/account_fr_fec_oca.py:
```diff
@@ -0,0 +1,579 @@
+            'property_account_receivable_id', 'res.partner')
+        return pay + rec
+
+    def do_query_unaffected_earnings(self):
```
il serait bien de mettre cette méthode en privée (du fait que c'est du pur SQL) + rajouter un `hasgroup`.

alexis-via replied:
Now that we have ir.model.access on wizards in v14, I think it's not usefull to have a hasgroup(), no ? I agree that it would be useful on v13-.

fmdl commented on l10n_fr_fec_oca/wizard/account_fr_fec_oca.py:
```diff
@@ -0,0 +1,579 @@
+        siren = vat[4:13] if not is_dom_tom else ''
+        return siren
+
+    def generate_fec(self):
```
idem, il serait bien de rajouter user.hasgroup(account_manager)

florian-dacosta commented on l10n_fr_fec_oca/wizard/account_fr_fec_oca.py:
```diff
@@ -0,0 +1,579 @@
+    partner_account_ids = fields.Many2many(
+        'account.account', string='Accounts',
+        default=lambda self: self._default_partner_account_ids())
+    fec_data = fields.Binary('FEC File', readonly=True, attachment=False)
```
@alexis-via I wonder, why `attachment=False` ?
I believe it should be changed. Indeed, I had an issue in version 9 with this error : 
`File "/odoo/src/openerp/sql_db.py", line 233, in executeres = self._obj.execute(query, params)InternalError: invalid memory alloc request size 1073741824`
When Odoo try to write the binary into the database.
It seems the file was too big for postgresql... Storing in in the filestore resolve the issue. 

Of course, it only happen when we have a lot of data. 
Can we consider changing this ?

alexis-via replied:
I've put attachment=False because it is like that in the native module, cf https://github.com/odoo/odoo/blob/14.0/addons/l10n_fr_fec/wizard/account_fr_fec.py#L20
Feel free to change it if you think it is better.

florian-dacosta replied:
Done !

florian-dacosta committed:
Run pre-commit

florian-dacosta committed:
Store attachment in filestore instead of db
The file can be huge in big databases and postgresql will then throw an error and can't process it.

fmdl commented:
FYI : https://github.com/odoo/odoo/pull/65170
